Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ExternalDB/ExternalDB.tsx

```diff
@@ -100,32 +100,35 @@ export default function ExternalDB(props: { ID: number, Tab: Tab }) {
 
     if (record == null) return null;
     return (
-        <div style={{ width: '100%', height: window.innerHeight - 63, maxHeight: window.innerHeight - 63, overflow: 'hidden', padding: 15 }}>
-            <LoadingScreen Show={requestStatus === 'loading'}/>
+        <div className="container-fluid d-flex h-100 flex-column" style={{ height: 'inherit' }}>
+            <LoadingScreen Show={requestStatus === 'loading'} />
             <div className="row">
-                <div className="col">
+                <div className="col-6 align-self-center">
                     <h2>{record?.Name ?? 'External Database'}</h2>
                 </div>
-                <div className="col">
+                <div className="col-6 align-self-center">
                     <button className="btn btn-danger pull-right" hidden={record == null}
                         onClick={() => setShowRemove(true)}>Delete External DB</button>
                     <button className="btn btn-light pull-right" hidden={record == null}
                         onClick={RequestUpdate}>Update Fields</button>
                 </div>
             </div>
 
-
-            <hr />
-            <TabSelector CurrentTab={tab} SetTab={(t: Tab) => setTab(t)} Tabs={Tabs} />
-
-            <div className="tab-content" style={{ maxHeight: window.innerHeight - 235 }}>
-                <div className={"tab-pane " + (tab == "info" ? " active" : "fade")} id="info">
-                    <ExternalDBInfo Record={record} />
-                </div>
-                <div className={"tab-pane " + (tab == "tables" ? " active" : "fade")} id="tables">
-                    <ExternalDBTables ID={record.ID} />
+            <div className="row" style={{ flex: 1, overflow: 'hidden' }}>
+                <div className="col-12" style={{ padding: 0 }}>
+                    <TabSelector CurrentTab={tab} SetTab={(t: Tab) => setTab(t)} Tabs={Tabs} />
+
+                    <div className="tab-content" style={{ height: 'calc(100% - 16px)' }}>
+                        <div className={"tab-pane " + (tab == "info" ? " active" : "fade")} id="info" style={{ height: 'inherit' }}>
```

Comment:
can we switch tab logic to `{tab == 'zyz' <div></div> : null}` for all tabs?
You can remove `id` and switch classes to `tab-pane active` if you do that. This causes issues because technically if you only mark tabs as `active` or not `active` when changing it does not reload the data once you switch tabs which may cause issues in some cases.